### ISSUE TITLE
blobs / unstructured data

### ISSUE THREAD

**Issue by awb99** [replikativ/datahike#314]
Is there a way to add a data blob / or fields that are not defined in the schema?
If there is no searching for this fields, this is no problem.

**Comment by den1k** [comment]
FYI, another OSS datalog database, [datalevin](https://github.com/juji-io/datalevin) can do this today

**Comment by awb99** [comment]
Thanks @den1k 
I started playing around with datalevin. But then went for datahike. 
My reason is that datahike has some documentation, and supports external databases.
I still feel tht datahike is quite experimental, and don't want to go too crazy for somethng sensible as a database

**Comment by whilo** [comment]
@awb99 If you just want to store blobs you can get the underlying konserve store out with `(:store @connection)` and then use the protocols for `edn` or binary data as described here: https://github.com/replikativ/konserve. You can also use konserve functions to access the blobs also from inside your queries, you will just not benefit from indexing.

**Comment by awb99** [comment]
Nice! Thanks @whilo  I have 25,000 documents of 5-10 MB each. I currently have them as individual edn files on disk, one by one.   Is this a problem for konserve?

**Comment by whilo** [comment]
No, this is no problem. One thing to keep in mind when using the same store as the database is that you should never mess with the key `:db` in there. All other values are random uuids and probably invisible to you anyway. A cleaner solution can be to use a separate store altogether, depending on whether this is possible for you or not.

**Comment by awb99** [comment]
@whilo Thanksfor your help!  This works for me. But let me ask again, because this is important. Why is it not possible to save fields that are not in the schema? Could you not just add :db.type/map or :db.type/vec and then you know that this is not 
searchable, but at least it can be retrieved with the same  logic. Perhaps this is what happens on schema-on-read?

In postgres I have blob. And so when I want to store a huge map, that does not need to be queried by postgres in the blob.
With datahike, as we are in clojure, instead of storing this type of data as a blob, it would just be stored as a map.

Am I wrong to think that this must be a very common usecase?

**Comment by whilo** [comment]
The problem with this is that we would overload the syntax for the edn-based transactor DSL, since maps and vectors in there have currently the meaning to automatically create new nested entities and triples that are automatically joined with the actual entity. What we can do is probably add a special keywords for literal values in the schema, e.g. `:datahike.edn/map` to pass the value through as a map. 

Having said this, you can already bypass values into the store by using the lower level triple notation, e.g. with `[128 :my-raw-map {...}]`, but I am not sure whether this plays well with all query functionality. Please try that out if you can and we can document this approach more openly and then think of the lifting to the transactor DSL. Does that work for you?
